src/main/java/com/alvarium/DefaultSdk.java

```diff
@@ -1,14 +1,25 @@
 package com.alvarium;
 
+import com.alvarium.annotators.Annotator;
+import com.alvarium.streams.StreamException;
+import com.alvarium.streams.StreamProvider;
+import com.alvarium.streams.StreamProviderFactory;
 import com.alvarium.utils.PropertyBag;
 
 public class DefaultSdk implements Sdk {
-
-  DefaultSdk() {
-    this.init();
-  }
-
-  private void init() {
+  private final Annotator[] annotators;
+  private final SdkInfo config;
+  private final StreamProvider stream;
+
+  // Logger should be implemented later
```

Comment:
Nitpick -- Put `TODO:` in front of this comment.

Perhaps also relevant -- I don't think we should implement our own logging provider as I've done with Go. I've used `log4j` in the past so you might want to take a look at that and see if it makes sense to extend your PR. There may be more current libraries to use, but at one time that was pretty ubiquitous. I know it uses the Apache 2.0 license, but check the transitive dependencies.

Reply:
I've taken a look at `log4j`, but it seems that it reached its EOL on 2015. Luckily, apache has release a successor called [`log4j2`](https://logging.apache.org/log4j/2.x/index.html), which is an upgrade to the `log4j` library. `log4j2` is split into two main parts: `API` and `core`, with each having its own dependency. the `API` part provides all the interfaces related to the logger and should be consumed by our sdk. On the other hand, the `core` part holds the actual implementation of the logger and shouldn't be a dependency in the production of our sdk. 

Following that, I've added the logger `API` as a compile dependency while the logger `core` as a test dependency in the maven `pom.xml` to be able to implement unit test for the sdk. The example app should provide the logger at runtime to the sdk through the constructor.

Reply:
added `log4j2` support in 95903542f5ea6de08b82708a1ee958b35018e4d6.